test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,264 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/netip"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher-sandbox/rancher-desktop-host-resolver/pkg/helper" //nolint:ignore
+	"github.com/rancher-sandbox/rancher-desktop-host-resolver/test/testdns"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/windows"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+)
+
+var (
+	// for now we use this maybe this can be an env var from test
+	wslDistroName = "Ubuntu"
+	testSrvAddr   = "127.0.0.1"
+	dnsPort       = "53"
+)
+
+func TestLookupARecords(t *testing.T) { //nolint:funlen
+	tmpDir, err := ioutil.TempDir("", "e2e_test_")
+	require.NoError(t, err, "Failed creating temp dir")
+	defer os.RemoveAll(tmpDir)
+
+	t.Logf("Creating %v wsl distro", wslDistroName)
+	installCmd := cmdExec("wsl", "--user", "root",
+		"--install",
+		"--distribution", wslDistroName)
```

Comment:
Yeah, the popup window is annoying and if we are considering disposing of the VM after since this will run on a github self-hosted runner, It may not be super bad. I considered the tarball, I just don't have a clear picture right now on how to produce the artifact, maybe we can visit this once we are setting up the CI.

Reply:
Do we have any particular needs from the distro? Because we _do_ have https://github.com/rancher-sandbox/rancher-desktop-wsl-distro just sitting there…